src/rpc/rawtransaction.cpp

```diff
@@ -349,19 +349,33 @@ UniValue getrawtransaction(const UniValue& params, bool fHelp)
     if (params.size() > 1)
         fVerbose = (params[1].get_int() != 0);
 
-    CTransaction tx;
+    // allocated by the callee
+    std::unique_ptr<CTransactionBase> pTxBase;
+    
     uint256 hashBlock;
-    if (!GetTransaction(hash, tx, hashBlock, true))
+    if (!GetTxBaseObj(hash, pTxBase, hashBlock, true) || !pTxBase)
```

Comment:
Can `GetTxBaseObj` return true and `pTxBase` be null? as far as I undestand it cannot happen, give current implementation.

Reply:
Given the current implementation this should never happen, but the caller should  always check the pTxBase anyway